library/waku_thread/waku_thread.nim

```diff
@@ -95,12 +95,16 @@ proc createWakuThread*(): Result[ptr Context, string] =
 
   return ok(ctx)
 
-proc stopWakuNodeThread*(ctx: ptr Context) =
+proc stopWakuNodeThread*(ctx: ptr Context): Result[void, string] =
   running.store(false)
+  let fireRes = ctx.reqSignal.fireSync()
```

Comment:
Please let me know if I understood correctly.

If we fire the signal, the thread moves from https://github.com/waku-org/nwaku/blob/9ef2eccb8aa19e80a9381dcb2ece29718ebce30f/library/waku_thread/waku_thread.nim#L59

And it will exit the loop in https://github.com/waku-org/nwaku/blob/9ef2eccb8aa19e80a9381dcb2ece29718ebce30f/library/waku_thread/waku_thread.nim#L55

Because of `running.load` not being `true` anymore?

Reply:
Yes, it should move from `waitFor ctx.reqSignal.wait()` and stop in the next line:
```
    let recvOk = ctx.reqChannel.tryRecv(request)
```
`recvOk` is false at this point, so the iteration will end and next loop will not be executed due to `running.load` being false.

I'm actually not sure if `tryRecv` will always be false. Perhaps I should change the code so it's like this? let me know what you think:
```nim
  while running.load == true:
    ## Trying to get a request from the libwaku main thread

    var request: ptr InterThreadRequest
    waitFor ctx.reqSignal.wait()
    if not running.load: break       <------------------------------
    let recvOk = ctx.reqChannel.tryRecv(request)
    if recvOk == true:

```

Reply:
Thanks for the explanation!

I personally like that alternative better :) Find it more predictable and easier to understand